src/Mvc/Mvc.Core/src/ModelBinding/Metadata/ValidationMetadata.cs

```diff
@@ -53,4 +53,9 @@ public class ValidationMetadata
     /// Gets or sets a value that determines if validators can be constructed using metadata on properties.
     /// </summary>
     internal bool PropertyHasValidators { get; set; }
+
+    /// <summary>
+    /// Gets or sets a model name that will be used in <see cref="ValidationEntry"/>.
+    /// </summary>
+    public string? ValidationModelName { get; set; }
```

Comment:
AFAIK this could be non-public.

Reply:
Let me double check but i think I changed to `public` because of the NewtsonftProvider.

Reply:
I double checked and we need it as public in order to update the property in the `NewtonsoftJsonMetadataProvider`. do you see a problem with that? I can investigate if I can use another approach.